### PR TITLE
CI: Checking in f-strings we use {repr(...)} instead of {...!r}

### LINKED ISSUE #29886: Replace "foo!r" to "repr(foo)" syntax
Since we are moving to f-strings and dropping old strings formats (```%``` and ```.format()```), we no longer need to have the redundant ```%r``` and ```!r``` specifiers, instead we use the ```repr()``` format.

#### Notes:
* Don't forget to link this issue in your PR, paste this ```https://github.com/pandas-dev/pandas/issues/29886``` in your PR.

* If any of your changed files are related to #29547 , please make sure to link your pull request to that issue as well, simply paste this ```https://github.com/pandas-dev/pandas/issues/29547``` in your pull request's body message as well.

* Please comment what you are planning to work on, so we won't do double work.

* If a file/files that should be marked as done, is'nt marked, please comment letting me know.

---

To check what files still needs to be fixed in the "pandas" directory:
```
grep -R -e '%r' -e '!r}' -e "'{}'" --include=*.{py,pyx} pandas/
```

##### Tip:
If you want to see the line number of the occurrence, replace the ```-l``` with ```-n```
for example:
```
grep -n -R '%r' --include=*.{py,pyx} pandas/
```

---

### The current list is:

- [x] ci/print_skipped.py

- [x] pandas/_config/config.py

- [x] pandas/conftest.py

- [x] pandas/core/accessor.py

- [x] pandas/core/algorithms.py

- [x] pandas/core/arrays/categorical.py

- [ ] pandas/core/arrays/datetimelike.py

- [ ] pandas/core/arrays/period.py

- [x] pandas/core/arrays/numpy_.py

- [x] pandas/core/computation/align.py

- [x] pandas/core/computation/eval.py

- [x] pandas/core/computation/expressions.py

- [x] pandas/core/computation/expr.py

- [x] pandas/core/computation/ops.py

- [x] pandas/core/computation/pytables.py

- [x] pandas/core/dtypes/common.py

- [x] pandas/core/dtypes/dtypes.py

- [x] pandas/core/frame.py

- [x] pandas/core/generic.py

- [x] pandas/core/groupby/grouper.py

- [x] pandas/core/indexes/base.py

- [x] pandas/core/indexes/datetimelike.py

- [x] pandas/core/indexes/interval.py

- [x] pandas/core/indexes/numeric.py

- [x] pandas/core/internals/concat.py

- [x] pandas/core/internals/managers.py

- [x] pandas/core/missing.py

- [x] pandas/core/nanops.py

- [x] pandas/core/reshape/concat.py

- [x] pandas/core/reshape/merge.py

- [x] pandas/core/reshape/tile.py

- [x] pandas/core/series.py

- [x] pandas/core/sorting.py

- [x] pandas/core/strings.py

- [x] pandas/core/window/rolling.py

- [x] pandas/\_\_init\_\_.py

- [x] pandas/io/formats/css.py

- [x] pandas/io/formats/excel.py

- [ ] pandas/io/formats/style.py

- [x] pandas/io/html.py

- [x] pandas/io/parsers.py

- [x] pandas/_libs/interval.pyx

- [x] pandas/plotting/_core.py

- [x] pandas/tests/computation/test_eval.py

- [x] pandas/tests/frame/test_alter_axes.py

- [x] pandas/tests/frame/test_analytics.py

- [x] pandas/tests/frame/test_query_eval.py

- [x] pandas/tests/groupby/test_whitelist.py

- [x] pandas/tests/indexes/datetimes/test_datetime.py

- [x] pandas/tests/indexes/multi/test_integrity.py

- [x] pandas/tests/indexes/period/test_period.py

- [x] pandas/tests/indexes/test_common.py

- [ ] pandas/tests/indexes/timedeltas/test_timedelta.py

- [x] pandas/tests/io/msgpack/test_case.py

- [x] pandas/tests/io/msgpack/test_extension.py

- [x] pandas/tests/io/parser/test_unsupported.py

- [x] pandas/tests/io/test_html.py

- [x] pandas/tests/test_expressions.py

- [x] pandas/tests/test_strings.py

- [ ] pandas/tests/io/test_sql.py

- [ ] pandas/tests/reshape/test_concat.py

- [ ] pandas/tests/scalar/timedelta/test_timedelta.py

- [x] pandas/tests/series/test_missing.py

- [x] pandas/tests/tseries/offsets/test_fiscal.py

- [x] pandas/tests/tseries/offsets/test_offsets.py

- [x] pandas/tseries/offsets.py

- [x] pandas/util/_decorators.py

- [x] pandas/util/_test_decorators.py

- [x] pandas/util/testing.py

- [x] pandas/_version.py

- [ ] /scripts/tests/test_validate_docstrings.py
---

#### Lastly:
- [ ] Add a test case (so we won't regress)

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref #29886
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

---

Can merge only after merging:

- [x] #29948 

- [x] #29950

- [x] #29953 

- [x] #29959

- [x] #29963

### CODE REVIEW

**Comment by WillAyd** [comment]
@MomIsBestFriend I think all precursors are done (?) can you merge master?

**Comment by MomIsBestFriend** [comment]
@WillAyd sorry for the delay, was on vacation.

Do you have an idea on how to exclude patterns like ```%%r```?

**Comment by jbrockmendel** [comment]
> Do you have an idea on how to exclude patterns like %%r?

You want to grep for "%r" but exclude "%%r" right?

With regular grep you could do `grep -R --include=*.{py,pyx} '%r' pandas | grep -v '%%r'`.  I'm not sure what it will take to make this work with the `invgrep` in the code checks

**Comment by MomIsBestFriend** [comment]
> > Do you have an idea on how to exclude patterns like %%r?
> 
> You want to grep for "%r" but exclude "%%r" right?
> 
> With regular grep you could do `grep -R --include=*.{py,pyx} '%r' pandas | grep -v '%%r'`. I'm not sure what it will take to make this work with the `invgrep` in the code checks

That's an interesting way of doing it, the problem is that the return code is messing my head. because the ```invgrep```  is giving the opposite of ```grep```. I just changed the matching pattern to make my life easier.

Hope my way of thinking is good.

**Comment by jbrockmendel** [comment]
@MomIsBestFriend pls rebase.  did you figure out the grep issue?

**Comment by MomIsBestFriend** [comment]
> did you figure out the grep issue?

Yes, I am greping for ```<Space>%r``` because it's more common, all the rest are edge cases, so it's not really worth writing test cases for them.

File: ci/code_checks.sh [hunk 1]
```diff
@@ -207,6 +207,14 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     invgrep -R --include=*.{py,pyx} 'xrange' pandas
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
+    MSG='Check for use of ("%r" % foo) instead of {repr(foo)}' ; echo $MSG
+    invgrep -R --include=*.{py,pyx} -E '\s%r' pandas | invgrep -v '%%r'
```
Comment (datapythonista) [inline ci/code_checks.sh]: The `\s` seems arbitrary. If there are cases of `%r` that we are happy to have, I think we should not add this rule.

Also, I don't understand the piped `invgrep`, but this looks too complex.

**Review by datapythonista — CHANGES_REQUESTED** [review]
Thanks for adding those @MomIsBestFriend 

The second rule looks ok, but I'm unsure about the first one.

File: ci/code_checks.sh (hunk 1 above)
Comment (MomIsBestFriend) [reply ci/code_checks.sh]: Agree, Will remove this rule.

**Comment by MomIsBestFriend** [comment]
Sorry for the short mess, messed something with the ```HEAD``` and the ```upstream/maser``` pointers.

It's all good now, correct?

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @MomIsBestFriend

**Comment by jreback** [comment]
thanks @MomIsBestFriend

**Pull request merged**